### PR TITLE
Introduce "minimal" collection

### PR DESCRIPTION
**Pull request by simonbaird**
[JIRA: HACBS-1615](https://issues.redhat.com/browse/HACBS-1615)

See commit messages for explanations.

### CODE REVIEW

**Comment by simonbaird** [comment]
[Docs preview here](http://file.bos.redhat.com/~sbaird/docs-demo/public/ec-policies/policy_configuration.html#_available_collections)

**Comment by simonbaird** [comment]
Todo:

* Fix tests
* Maybe move collection rule lists to somewhere else in the docs

File: Makefile [hunk 1]
```diff
@@ -138,15 +138,14 @@ HACBS_DOCS_REPO=git@github.com:hacbs-contract/hacbs-contract.github.io.git
...
+CURRENT_BRANCH=$(shell git rev-parse --abbrev-ref HEAD)
 docs-preview: $(HACBS_DOCS_DIR) ## Build a preview of the documentation
 	cd antora/ec-policies-antora-extension && \
 	  npm ci
 	cd $(HACBS_DOCS_DIR) && \
-	  sed -i 's|url: https://github.com/hacbs-contract/ec-policies.git|url: ../ec-policies|' antora-playbook.yml && \
-	  sed -i "s|require: '@hacbs-contract/ec-policies-antora-extension'|require: ../ec-policies/antora/ec-policies-antora-extension|" antora-playbook.yml && \
+	  yq e -i '.content.sources[] |= select(.url == "*ec-policies*").url |= "../ec-policies"' antora-playbook.yml && \
+	  yq e -i '.content.sources[] |= select(.url == "*ec-policies*").branches |= "$(CURRENT_BRANCH)"' antora-playbook.yml && \
+	  yq e -i '.antora.extensions[] |= select(.require == "*ec-policies-antora-extension").require |= "../ec-policies/antora/ec-policies-antora-extension"' antora-playbook.yml && \
```
Comment (zregvart) [inline Makefile:148]: I like this, could be a PR on it's own.

File: antora/docs/modules/ROOT/templates/_rule_links.hbs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Rules included:
+
+{{#each this}}
+// Fixme: This link wouldn't work if used for pipeline rule collections
```
Comment (zregvart) [inline antora/docs/modules/ROOT/templates/_rule_links.hbs:4]: Just for my understanding, this is because pipeline rules don't have collections? We don't use this for pipeline rules from what I can see, only in `policy_configuration.hbs`, so do we care? Should this be:

```suggestion
{{! This partial requires rule collection input }}
```

File: antora/docs/modules/ROOT/templates/policy_configuration.hbs [hunk 3]
```diff
@@ -180,20 +180,27 @@ file:
...
+=== Available collections
 
 [cols="2,6"]
 |===
 |*Name*
 |*Description*
 
-| `slsa_1`
-| Includes policy rules required to meet SLSA Level 1
+| `minimal`
```
Comment (zregvart) [inline antora/docs/modules/ROOT/templates/policy_configuration.hbs:190]: This section could also be generated if `processCollectionLists` were to return an array and we'd have a `[minimal|slsa1|slsa2|slsa3].rego` files containing the descriptions on the package.

File: antora/ec-policies-antora-extension/index.js [hunk 4]
```diff
@@ -114,10 +119,11 @@ const helpers = {
           const title = a.annotations.title
           const description = a.annotations.description
           const shortName = a.annotations.custom.short_name
-          const anchor = `${a.path.slice(1).map(e => e.value).join('_')}_${shortName}`
+          const anchor = `${ helpers.toJoinedPath(a.path.slice(3, a.path.length-1), "_") }__${ shortName }`
```
Comment (zregvart) [inline antora/ec-policies-antora-extension/index.js:122]: Could be in it's own PR

**Review by zregvart — COMMENTED** [review]
I like this, the side effect of having the collections in error codes is useful

File: Makefile (hunk 1 above)
Comment (simonbaird) [reply Makefile:148]: Actually I realized it's mostly duplicating `hack/local-build.sh` in the main docs repo...

Comment (simonbaird) [reply Makefile:148]: It does get its own commit.

File: antora/docs/modules/ROOT/templates/_rule_links.hbs (hunk 2 above)
Comment (simonbaird) [reply antora/docs/modules/ROOT/templates/_rule_links.hbs:4]: It's because the link hard codes the `release_policy.adoc` file name. If we ever did have collections for the pipeline rules, (which is a reasonable thing we might want to do), then link would be broken. (Actually I just thought of a way to fix this, maybe I'll try it.)

File: antora/docs/modules/ROOT/templates/policy_configuration.hbs (hunk 3 above)
Comment (simonbaird) [reply antora/docs/modules/ROOT/templates/policy_configuration.hbs:190]: Yeah, I would like to have new collections just appear in this list. I think it's doable, but will tackle it in a later PR I think.

Using a dummy rego package just for the collection descriptions annotations is an interesting new (to me at least) idea.

File: antora/docs/modules/ROOT/templates/_rule_links.hbs (hunk 2 above)
Comment (simonbaird) [reply antora/docs/modules/ROOT/templates/_rule_links.hbs:4]: Fixed in https://github.com/hacbs-contract/ec-policies/pull/295 .

**Comment by simonbaird** [comment]
I pulled out the misc fixes into other PRs.

Still draft though.

**Review by lcarva — APPROVED** [review]

File: antora/docs/modules/ROOT/templates/policy_configuration.hbs (hunk 3 above)
Comment (simonbaird) [reply antora/docs/modules/ROOT/templates/policy_configuration.hbs:190]: I filed https://issues.redhat.com/browse/HACBS-1749 about this.

**Review by zregvart — APPROVED** [review]
